crates/pallet-subspace-mmr/src/lib.rs

```diff
@@ -28,22 +29,42 @@ use sp_subspace_mmr::{LeafDataV0, MmrDigest, MmrLeaf};
 
 #[frame_support::pallet]
 mod pallet {
+    use frame_support::pallet_prelude::*;
     use frame_support::Parameter;
+    use frame_system::pallet_prelude::BlockNumberFor;
     use sp_core::H256;
 
     #[pallet::pallet]
     pub struct Pallet<T>(_);
 
     #[pallet::config]
     pub trait Config: frame_system::Config<Hash: Into<H256> + From<H256>> {
-        type MmrRootHash: Parameter + Copy;
+        type MmrRootHash: Parameter + Copy + MaxEncodedLen;
+
+        /// The number of mmr root hash store in runtime, it will be used to verify mmmr
+        /// proof stateless, the number of roots stored here represent the number of blocks
+        /// for which the mmr proof is valid since it is generated, after that the mmr proof
+        /// will be expired and the prover need to re-generate the proof.
+        type MmrRootHashCount: Get<u32>;
     }
+
+    /// Map of block numbers to mmr root hashes.
+    #[pallet::storage]
+    #[pallet::getter(fn mmr_root_hash)]
+    pub type MmrRootHashes<T: Config> =
+        StorageMap<_, Twox64Concat, BlockNumberFor<T>, T::MmrRootHash, OptionQuery>;
 }
 
 impl<T: Config> OnNewRoot<T::MmrRootHash> for Pallet<T> {
     fn on_new_root(root: &T::MmrRootHash) {
         let digest = DigestItem::new_mmr_root(*root);
         <frame_system::Pallet<T>>::deposit_log(digest);
```

Comment:
maybe we should remove this per our discussion since this is unneeded at this time
cc: @nazar-pc

Reply:
I believe this is already deployed to 3h so will leave a TODO for now.